### PR TITLE
docs: warning that runtime checks are incompatible with DefaultRouter…

### LINKED ISSUE #2016: strictStateImmutability + RouterState.Full incompatible with each other
Enabling `strictStateImmutability` together with `RouterState.Full` as router state serialization leads to errors that are very hard to track down (see angular/angular#31523, ngrx/platform#1937), so the [runtime check documentation](https://ngrx.io/guide/store/configuration/runtime-checks) should probably include a warning that those to configurations are not compatible with each other.

And for future (major) releases the `Minimal` router serialization should maybe be the default...?

## I would be willing to submit a PR for the docs :heart:

[X] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by luchsamapparat**
…StateSerializer

fixes #2016

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md [hunk 1]
```diff
@@ -31,6 +31,12 @@ It's possible to turn on the runtime checks one by one. To do so, you must enabl
 export class AppModule {}
 ```
 
+<div class="alert is-important">
+
+Runtime checks cannot be enabled if you use `@ngrx/router-store` with the `DefaultRouterStateSerializer`. The [default serializer](guide/router-store/configuration) returns a router state that is not serializable and contains references on components which should not be immutable. In order to use runtime checks either use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]:
```suggestion
Runtime checks cannot be enabled if you use `@ngrx/router-store` with the `DefaultRouterStateSerializer`. The [default serializer](guide/router-store/configuration) returns a router state that is not serializable and contains references on components which should not be immutable. To use runtime checks either use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
```

**Comment by luchsamapparat** [comment]
Actually, with Angular Ivy the immutability checks are also an issue. The full router state contains references to component classes. Making the state immutable means, that the component classes are readonly, which leads to this error:

```
core.js:7187 ERROR Error: Uncaught (in promise): TypeError: Cannot assign to read only property 'ngPrivateData' of function 'function HomepageComponent_Template(rf,ctx) {
      if (rf & 1) {
        jit___elementStart_4(0,'h1',_c0);
   ...<omitted>... }'
TypeError: Cannot assign to read only property 'ngPrivateData' of function 'function HomepageComponent_Template(rf,ctx) {
      if (rf & 1) {
        jit___elementStart_4(0,'h1',_c0);
   ...<omitted>... }'
    at getOrCreateTView (core.js:12110)
    at createRootComponentView (core.js:18020)
    at ComponentFactory$1.create (core.js:25107)
    at ViewContainerRef_.createComponent (core.js:20149)
    at RouterOutlet.activateWith (router.js:7330)
    at ActivateRoutes.activateRoutes (router.js:3160)
    at router.js:3097
    at Array.forEach (<anonymous>)
    at ActivateRoutes.activateChildRoutes (router.js:3092)
    at ActivateRoutes.activateRoutes (router.js:3167)
    at resolvePromise (zone-evergreen.js:797)
    at resolvePromise (zone-evergreen.js:754)
    at zone-evergreen.js:858
    at ZoneDelegate.invokeTask (zone-evergreen.js:391)
    at Object.onInvokeTask (core.js:30893)
    at ZoneDelegate.invokeTask (zone-evergreen.js:390)
    at Zone.runTask (zone-evergreen.js:168)
    at drainMicroTaskQueue (zone-evergreen.js:559)
defaultErrorLogger @ core.js:7187
```

I'd really recommend to change the default configuration in the next major release of @ngrx/router-store to use the minimal router serializer by default.

**Comment by brandonroberts** [comment]
@luchsamapparat we are considering that for version 9, but it breaks libraries like Nx that rely on the component https://github.com/nrwl/nx/blob/master/packages/angular/src/runtime/nx/data-persistence.ts#L426

**Comment by vicki2576** [comment]
Thank you

A-"74=HiwB8-^=d

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md [hunk 2]
```diff
@@ -31,6 +31,12 @@ It's possible to turn on the runtime checks one by one. To do so, you must enabl
 export class AppModule {}
 ```
 
+<div class="alert is-important">
+
+Runtime checks cannot be enabled if you use `@ngrx/router-store` with the `DefaultRouterStateSerializer`. The [default serializer](guide/router-store/configuration) returns a router state that is not serializable and contains references on components which should not be immutable. To use runtime checks either use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]:
```suggestion
The serializability runtime checks cannot be enabled if you use `@ngrx/router-store` with the `DefaultRouterStateSerializer`. The [default serializer](guide/router-store/configuration) has a unserializable router state and actions that are not serializable. To use the serializability runtime checks either use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
This also applies to Ivy with immutability runtime checks. 
```

File: projects/ngrx.io/content/guide/router-store/configuration.md [hunk 3]
```diff
@@ -18,6 +18,12 @@ interface StoreRouterConfig {
 
 If no router state serializer is provided through the [configuration](#configuration-options) of router store, the `DefaultRouterStateSerializer` is used. This router state serializer, serializes the URL together with the [ActivatedRouteSnapshot](https://angular.io/api/router/ActivatedRouteSnapshot) from [Angular Router](https://angular.io/guide/router). The latter is serialized recursively, but only with the possibility to traverse the route downward since `root` and `parent` parameters are set to `undefined`.
 
+<div class="alert is-important">
+
+The `DefaultRouterStateSerializer` cannot be used when [runtime checks](guide/store/configuration/runtime-checks) are enabled. If you want to use runtime checks to enforce serializability and immutability of your state, you can configure `RouterStoreModule` to use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/router-store/configuration.md]:
```suggestion
The `DefaultRouterStateSerializer` cannot be used when [serializability runtime checks](guide/store/configuration/runtime-checks) are enabled. If you want to use runtime checks to enforce serializability of your state and actions, you can configure `RouterStoreModule` to use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
This also applies to Ivy with immutability runtime checks.
```

**Pull request merged**